types/cli-table2/cli-table2-tests.ts

```diff
@@ -0,0 +1,232 @@
+import * as Table from "cli-table2";
+
+// Code snippets from README.md
+
+// Horizontal table
+const table1 = new Table({ head: ['TH 1 label', 'TH 2 label'], colWidths: [100, 200] }) as Table.HorizontalTable;
+table1.push(
+    ['First value', 'Second value'],
+    ['First value', 'Second value']
+);
+
+// Vertical table
+const table2 = new Table() as Table.VerticalTable;
```

Comment:
This is more of a convenience to restrict the types of arguments that can be passed to the array member functions.

The return value of the constructor without cast is basically of type:
```typescript
let table: Array<HorizontalTableRow> | VerticalTableRow | CrossTableRow;
```
which results in a push method having type:
```typescript
push: ((...items: HorizontalTableRow[]) => number) | ((...items: VerticalTableRow[]) => number) | ((...items: CrossTableRow[]) => number)
```
and passing a value of one of the three element types above gives:

> Cannot invoke an expression whose type lacks a call signature. Type '((...items: HorizontalTableRow[]) => number) | ((...items: VerticalTableRow[]) => number) | ((...items: Cross...' has no compatible call signatures.

instead of what i wanted it to be:
```typescript
push: (...items: HorizontalTableRow[] | VerticalTableRow[] | CrossTableRow[]) => number
```